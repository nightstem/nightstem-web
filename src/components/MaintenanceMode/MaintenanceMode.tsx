export const MaintenanceMode = () => {
  const version = process.env.NEXT_PUBLIC_APP_VERSION;

  return (
    <main className="relative w-dvw h-dvh bg-black overflow-hidden text-white grid place-content-center sm:items-center">
      <article className="relative flex flex-col gap-5 items-center text-center">
        <header className="flex flex-col text-center">
          <img
            src={'/logo.webp'}
            alt="logo"
            width={32}
            height={32}
            className="self-end animate-pulse"
          />
          <h1 className="heading-lg">Nightstem</h1>
        </header>

        <span className="text-neutral-500 text-sm tracking-wide">
          This page is under construction
        </span>
      </article>

      <span className="absolute bottom-4 right-4 text-xs text-neutral-400 opacity-60 font-mono">
        v{version}
      </span>
    </main>
  );
};

MaintenanceMode.displayName = 'MaintenanceMode';
export default MaintenanceMode;
