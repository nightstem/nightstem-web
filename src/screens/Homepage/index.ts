export { default as Homepage } from '@/screens/Homepage/Homepage';
