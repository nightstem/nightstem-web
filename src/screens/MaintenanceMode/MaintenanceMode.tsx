import Logo, {
  LOGO_ANIMATIONS,
  LOGO_MODES,
  LOGO_SIZES,
} from '@/components/ui/Logo';
import { LinkButton } from '@/components/ui/Buttons';

const MaintenanceMode = () => {
  const version = process.env.NEXT_PUBLIC_APP_VERSION;

  return (
    <div className="relative grid min-h-dvh w-full place-items-center">
      <section
        aria-labelledby="mm-title"
        aria-describedby="mm-desc"
        className="grid w-full max-w-prose gap-8 px-6"
      >
        <header className="grid gap-2">
          <Logo
            mode={LOGO_MODES.HORIZONTAL}
            size={LOGO_SIZES.SM}
            animation={LOGO_ANIMATIONS.PULSE}
            aria-label="Nightstem"
          />

          <h1 id="mm-title" className="heading-lg">
            We&apos;ll be back soon
          </h1>
        </header>

        <p id="mm-desc" className="text-foreground/70">
          We&apos;re making quiet upgrades. Thanks for your patience.
        </p>

        <div className="flex gap-3">
          <LinkButton href="." isExternal>
            Try again
          </LinkButton>
        </div>

        <div className="flex items-center gap-1.5 border-t border-foreground/10 pt-3 text-caption text-foreground/60">
          <span>Service unavailable (503)</span>
          <span>â€¢</span>
          <span>Please try again shortly.</span>
        </div>
      </section>

      {version && (
        <span className="absolute right-4 bottom-4 font-mono text-caption text-foreground/60">
          v{version}
        </span>
      )}
    </div>
  );
};

MaintenanceMode.displayName = 'MaintenanceMode';
export default MaintenanceMode;
