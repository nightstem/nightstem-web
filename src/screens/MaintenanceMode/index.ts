export * from '@/screens/MaintenanceMode/MaintenanceMode';
export { default } from '@/screens/MaintenanceMode/MaintenanceMode';
export * from '@/screens/MaintenanceMode/MaintenanceModeWrapper';
