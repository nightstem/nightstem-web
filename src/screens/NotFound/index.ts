export * from '@/screens/NotFound/NotFound';
export { default } from '@/screens/NotFound/NotFound';
