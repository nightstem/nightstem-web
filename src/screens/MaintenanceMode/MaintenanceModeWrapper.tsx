'use client';

import { ReactNode } from 'react';
import { MaintenanceMode } from './MaintenanceMode';

type MaintenanceModeWrapperProps = {
  children: ReactNode;
  isMaintenanceModeOn?: boolean;
};

export const MaintenanceModeWrapper = ({
  children,
  isMaintenanceModeOn = false,
}: MaintenanceModeWrapperProps) => {
  return isMaintenanceModeOn ? <MaintenanceMode /> : <>{children}</>;
};
