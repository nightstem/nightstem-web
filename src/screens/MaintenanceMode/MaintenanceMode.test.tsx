import { axe } from 'vitest-axe';
import { render } from '@testing-library/react';

import { MaintenanceMode } from '@/screens/MaintenanceMode';

const mockVersion = '1.0.0';
const originalEnv = process.env;

beforeEach(() => {
  process.env = {
    ...originalEnv,
    NEXT_PUBLIC_APP_VERSION: mockVersion,
  };
});

afterEach(() => {
  process.env = originalEnv;
});

describe('MaintenanceMode', () => {
  it('should pass accessibility tests', async () => {
    const { container } = render(<MaintenanceMode />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });

  it('should match snapshot', () => {
    const { container } = render(<MaintenanceMode />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
