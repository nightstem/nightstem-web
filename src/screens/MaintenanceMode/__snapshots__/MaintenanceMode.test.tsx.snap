// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MaintenanceMode > should match snapshot 1`] = `
<div
  class="relative grid min-h-dvh w-full place-items-center"
>
  <section
    aria-describedby="mm-desc"
    aria-labelledby="mm-title"
    class="grid w-full max-w-prose gap-8 px-6"
  >
    <header
      class="grid gap-2"
    >
      <div
        class="inline-flex items-center gap-2"
      >
        <svg
          aria-hidden="true"
          class="text-primary-500 h-[1.066rem] w-[1.066rem] animate-pulse motion-reduce:animate-none shrink-0"
          fill="none"
          viewBox="0 0 500 500"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M346.04 135.875L500 245.427L161.185 495.252C157.629 497.874 151.828 497.923 148.228 495.362L0.786259 390.448L346.04 135.875Z"
            fill="currentColor"
          />
          <path
            d="M148.322 4.74771C151.878 2.12562 157.68 2.07657 161.279 4.63814L333.785 127.386L179.023 241.499L9.54552e-05 114.113L148.322 4.74771Z"
            fill="currentColor"
          />
        </svg>
        <span
          class="heading-sm"
        >
          Nightstem
        </span>
      </div>
      <h1
        class="heading-lg"
        id="mm-title"
      >
        We'll be back soon
      </h1>
    </header>
    <p
      class="text-foreground/70"
      id="mm-desc"
    >
      We're making quiet upgrades. Thanks for your patience.
    </p>
    <div
      class="flex gap-3"
    >
      <a
        class="select-none whitespace-nowrap font-medium cursor-pointer disabled:cursor-not-allowed transition-colors duration-200 motion-reduce:transition-none focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 ring-offset-dark text-base px-4 py-2 rounded-md text-dark bg-primary-500 hover:bg-primary-400 active:bg-primary-400 focus-visible:ring-primary-500 disabled:opacity-60 disabled:hover:bg-primary-500 disabled:active:bg-primary-500"
        href="."
      >
        Try again
      </a>
    </div>
    <div
      class="flex items-center gap-1.5 border-t border-foreground/10 pt-3 text-caption text-foreground/60"
    >
      <span>
        Service unavailable (503)
      </span>
      <span>
        â€¢
      </span>
      <span>
        Please try again shortly.
      </span>
    </div>
  </section>
  <span
    class="absolute right-4 bottom-4 font-mono text-caption text-foreground/60"
  >
    v
    1.0.0
  </span>
</div>
`;
